src/registrar/assets/js/get-gov-admin.js

```diff
@@ -296,46 +296,130 @@ function initializeWidgetOnList(list, parentId) {
 }
 
 /** An IIFE for admin in DjangoAdmin to listen to changes on the domain request
- * status select and to show/hide the rejection reason
+ * status select and to show/hide fields like rejection reason or action needed reason
 */
 (function (){
-    let rejectionReasonFormGroup = document.querySelector('.field-rejection_reason')
-
-    if (rejectionReasonFormGroup) {
-        let statusSelect = document.getElementById('id_status')
-
-        // Initial handling of rejectionReasonFormGroup display
-        if (statusSelect.value != 'rejected')
-            rejectionReasonFormGroup.style.display = 'none';
+    // Hides or shows a given field based off of the current value of the given status selector,
+    // and stores its state in the session.
+    function showHideFieldsOnStatusChange(elementToHide, statusToShowOn, sessionObjectName) {
+        if (elementToHide) {
+            let statusSelect = document.getElementById('id_status')
+            
+            let shouldHide = statusSelect.value != statusToShowOn
+            // Initial handling of parentFormGroup display
+            hideOrShowDomObject(elementToHide, hideObject=shouldHide)
+    
+            // Listen to change events and handle rejectionReasonFormGroup display, then save status to session storage
+            statusSelect.addEventListener('change', function() {
+                // Hide the object only if we're in an invalid state
+                shouldHide = statusSelect.value != statusToShowOn
+
+                // Hide the action needed field if we're on a different status type
+                hideOrShowDomObject(elementToHide, hideObject=shouldHide)
+
+                // Add a key to our session storage to track if we should hide the object automatically
+                // (to catch the edge case where you click the back button)
+                if (!shouldHide){
+                    sessionStorage.removeItem(sessionObjectName);
+                }else {
+                    sessionStorage.setItem(sessionObjectName, 'true');
+                }
+            });
+        }
+    }
 
-        // Listen to change events and handle rejectionReasonFormGroup display, then save status to session storage
-        statusSelect.addEventListener('change', function() {
```

Comment:
Why not simply expand the old existing code with a 3rd option: value == 'change_requested"? I think this could be simpler. If in the future more statuses will require reasons, we could refactor then.

Reply:
Looks good!